x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -76,6 +98,24 @@ export const CreateField = React.memo(function CreateFieldComponent({
     }
   };
 
+  const [referenceFieldComboValue, setReferenceFieldComboValue] = useState<any>();
```

Comment:
Same with these four values: don't use `any`, define a type. Defining a type also means you won't need to do so many checks further down this component.